src/utils/clmfates_interfaceMod.F90

```diff
@@ -1106,6 +1109,33 @@ subroutine wrap_update_hlmfates_dyn(this, nc, bounds_clump,      &
             this%f2hmap(nc)%fcolumn, &
             this%fates(nc)%bc_out )
 
+       !------------------------------------------------------------------------
+       ! FATES calculation of ligninNratio
+       !------------------------------------------------------------------------
+       ! If it's the first timestep of a restart and copy_fates_var(c) = .false.
+       ! (this will happen in the first timestep of any restart)
+       ! then skip this variable because a more accurate value was obtained
+       ! from the restart file.
+       ! Note 1. I had hoped is_first_restart_step() alone would suffice, but
+       ! is_first_restart_step() remained true for the whole first day in my
+       ! test. Hence I added the check for copy_fates_var(c) which changes to
+       ! .true. the first time that we come through this code in a restart.
+       ! Note 2. copy_fates_var is a column-level array to make thread-safe.
+       ! Note 3. This if statement is a workaround for restart tests to pass.
+       ! Ideally, the fates variable would have the correct value at restart,
+       ! but rgknox explains that accomplishing this is more complex given the
+       ! current FATES treatment of other litter fluxes passed to the CTSM.
+       !------------------------------------------------------------------------
+       do s = 1, this%fates(nc)%nsites
+          c = this%f2hmap(nc)%fcolumn(s)
+          if (is_first_restart_step() .and. .not. copy_fates_var(c)) then
+             copy_fates_var(c) = .true.
```

Comment:
Moved the if statement inside the column loop now.

Reply:
@slevisconsulting I'm not sure we should add this to the test suite. But, I would like you to run a threaded test for this. So run

ERP_P144x2_Ld30.f45_f45_mg37.I2000Clm50FatesRs.cheyenne_intel.clm-mimicsFatesColdDef

It looks like currently the only threaded FATES test we run is:

ERP_D_P32x2_Ld3.f19_g17.I2000Clm50FatesCru.cheyenne_intel.clm-FatesColdDef

But, it is passing. We should likely add some more...

Reply:
Good call @ekluzek 
this threaded test fails, so I'm investigating.

Reply:
You might try some other threaded tests to see what works and what breaks. I'd try both changing the above ERP to ERS and SMS. And you should try some ones that are closer to the test that works in the baseline. So try the baseline test of:

ERP_D_P32x2_Ld3.f19_g17.I2000Clm50FatesCru.cheyenne_intel.clm-FatesColdDef

and then also do...

ERP_D_P32x2_Ld3.f19_g17.I2000Clm50FatesCru.cheyenne_intel.clm-mimicsFatesColdDef

That should help to figure out if it's threading and FATES in general or if it's just in MIMICS/FATES. And the SMS, ERS, and ERP tests tell if it's a problem with threading in general, with restarts in general, or only when changing PE's on restart. All of that would be good to know.